### PR TITLE
Default to thin pool queueing and allow changing behavior

### PR DESCRIPTION
**Pull request by jbaublitz**
Related to stratis-storage/stratisd#2814

### CODE REVIEW

**Comment by jbaublitz** [comment]
Blocked by #737

File: src/thinpooldev.rs [hunk 1]
```diff
@@ -508,6 +511,32 @@ impl ThinPoolDev {
...
+    /// where the thin-pool-specific string has the format:
+    /// <meta maj:min> <data maj:min> <block size> <low water mark>
+    /// There is exactly one entry in the table.
+    fn gen_table(
+        meta: &LinearDev,
+        data: &LinearDev,
+        data_block_size: Sectors,
+        low_water_mark: DataBlocks,
+        feature_args: Vec<String>,
+    ) -> ThinPoolDevTargetTable {
```
Comment (mulkieran) [inline src/thinpooldev.rs:526]: Should probably have ```gen_default_table``` invoke ```gen_table```.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Code looks good, one request. The idea of toggling ```error_if_no_space``` is a little odd to me, I would have felt more confident with methods to set and unset it (which would have been pretty minimalist). But, go w/ toggle if you're sure that's what you want. I already looked at the source in stratisd and I understand the usage.

**Comment by jbaublitz** [comment]
> Code looks good, one request. The idea of toggling error_if_no_space is a little odd to me, I would have felt more confident with methods to set and unset it (which would have been pretty minimalist). But, go w/ toggle if you're sure that's what you want. I already looked at the source in stratisd and I understand the usage.

You may have a point here mainly because I think it makes a lot of sense from the point of view of stratisd as a consumer, but maybe less sense as a general solution in a library. Let me play around with this and see what the alternative looks like.

File: src/thinpooldev.rs (hunk 1 above)
Comment (jbaublitz) [reply src/thinpooldev.rs:526]: Resolved!

**Review by mulkieran — APPROVED** [review]

**Comment by mulkieran** [comment]
@jbaublitz Plz squash and merge Thursday.

**Pull request merged**